drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c

```diff
@@ -2351,13 +2375,22 @@ static int bcm2835_codec_open(struct file *file)
 					 BIT(V4L2_MPEG_VIDEO_H264_PROFILE_MAIN) |
 					 BIT(V4L2_MPEG_VIDEO_H264_PROFILE_HIGH)),
 					V4L2_MPEG_VIDEO_H264_PROFILE_HIGH);
-		if (hdl->error) {
-			rc = hdl->error;
-			goto free_ctrl_handler;
-		}
-		ctx->fh.ctrl_handler = hdl;
-		v4l2_ctrl_handler_setup(hdl);
+	} else {
```

Comment:
Needs to be 
```
else if (dev->role == DECODE) {
```
otherwise you are adding this control for the ISP role as well.

Reply:
> bcm2835-codec is a little different from other decoder implementations in that there is an intermediate format conversion between the hardware and V4L2 buffers. The number of capture buffers required is therefore independent of the stream and DPB etc.
> 
> There are plans to remove the conversion, but it requires a fair amount of rework within the firmware. Until that is done, you may as well return a static value of 1.

I see. May I quote the above when I rework the commit log, or should I just drop the commit for now?

> https://hverkuil.home.xs4all.nl/codec-api/uapi/v4l/dev-decoder.html is Hans hosting the latest copy of the decoder spec - no mention of `V4L2_CID_MIN_BUFFERS_FOR_CAPTURE` being mandatory.

I must have misread something. I'll drop that part.

Reply:
> I see. May I quote the above when I rework the commit log, or should I just drop the commit for now?

Sure.

I'm happy for it to be implemented as a control if it makes more clients magically work.
Checking it in https://hverkuil.home.xs4all.nl/codec-api/uapi/v4l/control.html
> This is a read-only control that can be read by the application and used as a hint to determine the number of CAPTURE buffers to pass to REQBUFS. The value is the minimum number of CAPTURE buffers that is necessary for hardware to work.

So the client is only meant to treat it as a hint, therefore failing should the query fail seems like an error in the client.